rust/datafusion/src/optimizer/utils.rs

```diff
@@ -24,7 +24,11 @@ use arrow::datatypes::{Schema, SchemaRef};
 use super::optimizer::OptimizerRule;
 use crate::error::{DataFusionError, Result};
 use crate::logical_plan::{Expr, LogicalPlan, PlanType, StringifiedPlan};
-use crate::prelude::col;
+use crate::prelude::{col, lit};
+use crate::scalar::ScalarValue;
+
+const CASE_EXPR_MARKER: &str = "__DATAFUSION_CASE_EXPR__";
```

Comment:
🤔

Reply:
I couldn't think of another way to do the expression rewriting with our current design. This expression is quite different from the ones implemented so far.